packages/core/src/browser/frontend-application.ts

```diff
@@ -25,6 +25,9 @@ import { FrontendApplicationStateService } from './frontend-application-state';
 import { preventNavigation, parseCssTime, animationFrame } from './browser';
 import { CorePreferences } from './core-preferences';
 import { WindowService } from './window/window-service';
+import { loadTranslations } from './nls';
+
+loadTranslations();
```

Comment:
I don't think we should be triggering an async operation at this point and then cross fingers that the rest of the modules will properly pick up what resolved here...

Here's a fix: https://github.com/msujew/theia/compare/i18n...eclipse-theia:mp/i18n

Note that I had to modify the monaco loader as it didn't like not running after the page was loaded (due to waiting for the translation request to go through).

Reply:
Thanks for the fix, I applied it to my branch and it works really well.